From 1e47ecd9a9367e6cf166c872d3cec26eb4c77aba Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Fri, 10 Nov 2017 19:51:57 +0100
Subject: [PATCH 12/13] console: report detach message on demand

When users pass -1 there's there won't be an escape sequence to exit the
console so no need to print a misleading info message about how to detach.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/console.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/lxc/console.c b/src/lxc/console.c
index e95f5c54..8f6203f0 100644
--- a/src/lxc/console.c
+++ b/src/lxc/console.c
@@ -730,11 +730,14 @@ int lxc_console(struct lxc_container *c, int ttynum,
 		goto close_mainloop;
 	}
 
-	fprintf(stderr, "\n"
+	if (ts->escape >= 1) {
+		fprintf(stderr,
+			"\n"
 			"Connected to tty %1$d\n"
 			"Type <Ctrl+%2$c q> to exit the console, "
 			"<Ctrl+%2$c Ctrl+%2$c> to enter Ctrl+%2$c itself\n",
 			ttynum, 'a' + escape - 1);
+	}
 
 	if (istty) {
 		ret = lxc_setup_tios(stdinfd, &oldtios);
-- 
2.11.0

